include/cmdlib/CommandFacility.hpp

```diff
@@ -87,7 +87,7 @@ class CommandFacility
   CompletionQueue m_completion_queue;
 
   //! Request callback function signature
-  typedef std::function<void(const cmdobj_t&)> CommandCallback;
+  typedef std::function<void(cmdmeta_t)> CommandCallback;
```

Comment:
Making a copy of huge configuration objects calling the async might affect performance. I would keep passing by ref.

Reply:
Fixed. Only `cmdmeta_t` is copied now.